nservicebus/upgrades/5to6/distributor.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Migrating the Distributor to use Sender Side Distribution
+tags:
+ - upgrade
+ - migration
+related:
+ - samples/scaleout/distributor-upgrade
+---
+
+The [distributor](/nservicebus/scalability-and-ha/distributor) has been deprecated and replaced by [sender-side distribution](/nservicebus/msmq/scalability-and-ha/sender-side-distribution.md) with NServiceBus Version 6.
+
+## Client-side distribution
+
+With sender-side distribution, senders are aware of scaled out receivers, allowing senders to distribute messages directly to the receivers instead of using an intermediary distributor.
+
+
+## General upgrade scenario
+
+This process aims to allow upgrade in production environments without missing any messages and minimal downtime. If a scaled-out endpoint only receives commands and does not subscribe to messages consider following the [Simple upgrade scenario](#simple-upgrade-scenario) instead.
+
+* Upgrade all endpoints that interact with the distributor to Version 6 first. At this stage, do not upgrade the workers to Version 6.
+  * upgrade all endpoints that send command messages to the Distributor endpoint to be distributed,
+  * upgrade all endpoints that send subscription messages to the Distributor, 
+  * upgrade all endpoints subscribing to events published by worker nodes.
+* Apply the following steps for each worker, one after another:
```

Comment:
If you migrate all the workers, how do you make sure no messages are left in the distributor queues? If you migrate all of them to not use the distributor, nobody will pick from the distributor queue. If you migrate some of them to use the distributor and some to not, the distributor queue will still be round robined to, meaning it will never empty out because there will be fresh incoming traffic continously.